app/controllers/pets_controller.rb

```diff
@@ -42,6 +47,7 @@ def adoption_status_adoptable
 
   def destroy
     Pet.destroy(params[:id])
+    favorites.delete_pet(params[:id])
```

Comment:
It could be argued that because this is affecting the Favorites DB, it should be handled by the model